File: mini-commerce/src/test/java/me/hero/minicommerce/acepptance/ItemSteps.java
package me.hero.minicommerce.acepptance;

import io.restassured.RestAssured;
import io.restassured.response.ExtractableResponse;
import io.restassured.response.Response;
import java.util.HashMap;
import java.util.Map;
import org.springframework.http.MediaType;

public class ItemSteps {
  public static ExtractableResponse<Response> 상품_생성(Map<String, String> params) {
    ExtractableResponse<Response> response = RestAssured.given().log().all()
        .body(params)
        .contentType(MediaType.APPLICATION_JSON_VALUE)
        .when()
        .post("/items")
        .then().log().all()
        .extract();
    return response;
  }

  public static Map<String, String> 상품_매개변수_생성(String name, String price) {
    Map<String, String> params = new HashMap<>();
    params.put("name", name);
    params.put("price", price);
    return params;
  }

  public static ExtractableResponse<Response> 상품_수정(String updateLocation,
      Map<String, String> updateParams) {
    ExtractableResponse<Response> response = RestAssured.given().log().all()
        .body(updateParams)
        .contentType(MediaType.APPLICATION_JSON_VALUE)
        .when()
        .put(updateLocation)
        .then().log().all()
        .extract();
    return response;
  }

  public static ExtractableResponse<Response> 상품_삭제(String deleteLocation) {
    return RestAssured.given().log().all()
        .contentType(MediaType.APPLICATION_JSON_VALUE)
        .when()
        .delete(deleteLocation)
        .then().log().all()
        .extract();
  }




}

File: mini-commerce/src/test/kotlin/me/hero/minicommerce/acepptance/ItemSteps.kt
package me.hero.minicommerce.acepptance

import io.restassured.RestAssured
import io.restassured.response.ExtractableResponse
import io.restassured.response.Response
import org.springframework.http.MediaType

object ItemSteps {
    fun 상품_생성(params: Map<String, String>): ExtractableResponse<Response> {
        return RestAssured.given().log().all()
            .body(params)
            .contentType(MediaType.APPLICATION_JSON_VALUE)
            .`when`()
            .post("/items")
            .then().log().all()
            .extract()
    }

    fun 상품_매개변수_생성(name: String, price: String): Map<String, String> {
        val params: MutableMap<String, String> = HashMap()
        params["name"] = name
        params["price"] = price
        return params
    }

    fun 상품_수정(
        updateLocation: String?,
        updateParams: Map<String, String>
    ): ExtractableResponse<Response> {
        return RestAssured.given().log().all()
            .body(updateParams)
            .contentType(MediaType.APPLICATION_JSON_VALUE)
            .`when`()
            .put(updateLocation)
            .then().log().all()
            .extract()
    }

    fun 상품_삭제(deleteLocation: String?): ExtractableResponse<Response> {
        return RestAssured.given().log().all()
            .contentType(MediaType.APPLICATION_JSON_VALUE)
            .`when`()
            .delete(deleteLocation)
            .then().log().all()
            .extract()
    }

    fun 상품_조회(deleteLocation: String?): ExtractableResponse<Response> {
        return RestAssured.given().log().all()
            .contentType(MediaType.APPLICATION_JSON_VALUE)
            .`when`()[deleteLocation]
            .then().log().all()
            .extract()
    }
}